DRAFT.md

```diff
@@ -892,26 +902,23 @@ constexpr indirect& operator=(indirect&& other) noexcept(
   `std::allocator_traits<Alloc>::propagate_on_container_move_assignment` is
   `true` and `alloc != other.alloc` then the allocator needs updating.
 
-  If `other` is valueless, `*this` becomes valueless and the owned value in this,
-  if any, is destroyed using `allocator_traits<allocator_type>::destroy` and then
-  deallocated using `allocator_traits<allocator_type>::deallocate`.
+  If `other` is valueless, `*this` becomes valueless and the owned object in
+  this, if any, is destroyed using `allocator_traits<allocator_type>::destroy`
+  and then the storage is deallocated.
 
-  Otherwise if `alloc == other.alloc`, swaps the owned objects in `this` and
+  Otherwise, if `alloc == other.alloc`, swaps the owned objects in `this` and
   `other`; the owned object in `other`, if any, is then destroyed using
-  `allocator_traits<allocator_type>::destroy` and then deallocated using
-  `allocator_traits<allocator_type>::deallocate`.
+  `allocator_traits<allocator_type>::destroy` and then the storage is deallocated.
 
-  Otherwise , if `alloc != other.alloc` or `this` is valueless, a new owned
+  Otherwise, if `alloc != other.alloc` or `this` is valueless, a new owned
```

Comment:
Actually, is the "or `*this` is valueless" condition unreachable? We have:

if (other is valueless) { }
else if (alloc == other.alloc) { }
else if (alloc != other.alloc || *this is valueless) { }

But we know by the time we get to the third `if` that alloc != other.alloc because we didn't take the second branch, so it can be simply:

if (other is valueless) { }
else if (alloc == other.alloc) { }
else { }

So the line above can be simple "Otherwise, a new owned ..."

Reply:
I think the same observation is true for the copy assignment operator, but at least there the "alloc != other.alloc or *this is not valueless" is the inverse of the previous condition. Here on the move assignment operator it looks like a copy & paste error because "or *this is valueless" makes no sense here.

Reply:
Agreed and fixed.